### DISCUSSION TITLE
How to use web3Auth react-native with backend apis

### DISCUSSION THREAD

**Discussion by BlockJedi** [Web3Auth/Web3Auth#629] [Help]
Hi, We are creating and app, want to use web3Auth social logins. We want to store some user information like name, email and account public key in mongodb. I am concerned about how we can use authentication on apis if we use web3Auth. Like user want to get some list of things on app his information is stored on web3auth, so how I restrict api to show his objects only.

**Comment by himanshuchawla009** [comment]
Hi, You can refer the server side verification mechanism described in link below, basically we return a idToken along with user info on frontend and you can validate that idToken on your backend to verify user's identity for api access and also decode that token to read user's profile information.

https://web3auth.io/docs/server-side-verification/social-login-users
